cadc-dali/src/main/java/ca/nrc/cadc/dali/tables/parquet/ParquetReader.java

```diff
@@ -0,0 +1,190 @@
+package ca.nrc.cadc.dali.tables.parquet;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.avro.generic.GenericRecord;
+
+import org.apache.log4j.Logger;
+
+import org.apache.parquet.avro.AvroParquetReader;
+import org.apache.parquet.hadoop.ParquetFileReader;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.apache.parquet.io.InputFile;
+import org.apache.parquet.io.SeekableInputStream;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.Type;
+
+
+//TODO: Temporary implementation
+public class ParquetReader {
+
+    private static final Logger log = Logger.getLogger(ParquetReader.class);
+
+    public TableShape read(InputStream inputStream) throws IOException {
+
+        InputFile inputFile = getInputFile(inputStream);
+        int recordCount = 0;
+        int columnCount = 0;
+
+        ParquetMetadata metadata = ParquetFileReader.open(inputFile).getFooter();
+        MessageType schema = metadata.getFileMetaData().getSchema();
+        columnCount = schema.getFieldCount();
+
+        String votable = metadata.getFileMetaData().getKeyValueMetaData().get("votable");
```

Comment:
You could read this xml string using VOTableReader and then add the VOTableDocument object to the TableShape object that is returned... we could then test that metadata round trip is correct.

I feel like the future of this read() method will be to return a class that holds various things... and the rich metadata will be one of those things. I'm not sure what form the returned data might take, so don't think we should do anything for that just yet.